src/hammer-vlsi/hammer_vlsi/hammer_tool.py

```diff
@@ -970,7 +973,7 @@ def get_gds_map_file(self) -> Optional[str]:
         :return: Fully-resolved path to GDS map file or None.
         """
         # Mode can be auto, empty, or manual
-        gds_map_mode = str(self.get_setting("par.inputs.gds_map_mode"))  # type: str
+        gds_map_mode = ModeType.from_str(str(self.get_setting("par.inputs.gds_map_mode")))  # type: ModeType
```

Comment:
We may want to create a static helper method called `ModeType.from_setting` that takes the `HammerTool` object and the setting string as args and returns a `ModeType`. e.g.

```
class ModeType:
  @staticmethod
  def from_setting(tool: HammerTool, setting: str) -> ModeType:
    return ModeType.from_str(str(tool.get_setting(setting)))
```
which would then allow the following change (no need for type hint, either):

```suggestion
        gds_map_mode = ModeType.from_setting(self, "par.inputs.gds_map_mode")
```

Reply:
Not to nitpick too much - at that point I would prefer almost `gds_map_mode = self.get_mode_setting("par.inputs.gds_map_mode")` to avoid a dependency from `ModeType` -> `HammerTool`

Reply:
I also like that, perhaps more.